test/csuite/wt7989_compact_checkpoint/main.c

```diff
@@ -0,0 +1,286 @@
+
+    /* Spawn checkpoint and compact threads. */
+    testutil_check(pthread_create(&thread_compact, NULL, thread_func_compact, &td));
+    testutil_check(pthread_create(&thread_checkpoint, NULL, thread_func_checkpoint, &td));
+    (void)pthread_join(thread_checkpoint, NULL);
+    (void)pthread_join(thread_compact, NULL);
+
+    file_sz_after = get_file_size(session, uri);
+
+    /* Cleanup */
+    if (!stress_test) {
+        __wt_cond_destroy(td.session_cond, &td.cond);
+        td.session_cond = NULL;
+        td.cond = NULL;
+    }
+
+    testutil_check(session->close(session, NULL));
+
+    /* Check if there's at least 10% compaction. */
+    testutil_assertfmt(file_sz_before * 0.9 > file_sz_after,
+      "Compact failed to reduce file size by at least 10%%\n"
+      "Compressed file size MB: %f\n"
+      "Original file size MB: %f\n",
+      file_sz_after / (1024.0 * 1024), file_sz_before / (1024.0 * 1024));
+}
+
+static void *
+thread_func_compact(void *arg)
+{
+    struct thread_data *td;
+    WT_SESSION *session;
+
+    td = (struct thread_data *)arg;
+
+    testutil_check(td->conn->open_session(td->conn, NULL, NULL, &session));
+
+    if (td->cond != NULL) {
+        /* Make sure checkpoint thread is initialized and waiting for the signal. */
+        __wt_sleep(1, 0);
+
+        /* Wake up the checkpoint thread. */
+        __wt_cond_signal(td->session_cond, td->cond);
+    }
+
+    /* Perform compact operation. */
+    testutil_check(session->compact(session, td->uri, NULL));
+    testutil_check(session->close(session, NULL));
+
+    return (NULL);
+}
+
+static void *
+thread_func_checkpoint(void *arg)
+{
+    struct thread_data *td;
+    WT_SESSION *session;
+    bool signalled;
+
+    td = (struct thread_data *)arg;
+
+    testutil_check(td->conn->open_session(td->conn, NULL, NULL, &session));
+
+    if (td->cond != NULL) {
+        __wt_cond_wait_signal(td->session_cond, td->cond, 0, NULL, &signalled);
+    }
+
+    testutil_check(session->checkpoint(session, NULL));
+    testutil_check(session->close(session, NULL));
+
+    return (NULL);
+}
+
+static void
+populate(WT_SESSION *session, const char *uri)
+{
+    WT_CURSOR *cursor;
+    time_t t;
+    uint64_t val;
+    int i, str_len;
+
+    srand((u_int)time(&t));
+
+    str_len = sizeof(data_str) / sizeof(data_str[0]);
+    for (i = 0; i < str_len - 1; i++) {
+        data_str[i] = 'a' + rand() % 26;
+    }
+    data_str[str_len - 1] = '\0';
+
+    testutil_check(session->open_cursor(session, uri, NULL, NULL, &cursor));
+    for (i = 0; i < NUM_RECORDS; i++) {
+        cursor->set_key(cursor, i);
+        val = (uint64_t)rand();
+        cursor->set_value(cursor, val, val, val, data_str);
+        testutil_check(cursor->insert(cursor));
+    }
+
+    testutil_check(cursor->close(cursor));
+}
+
+static void
+remove_records(WT_SESSION *session, const char *uri)
+{
+    WT_CURSOR *cursor;
+    int i;
+
+    testutil_check(session->open_cursor(session, uri, NULL, NULL, &cursor));
+
```

Comment:
Add a comment about the range of keys being deleted (i.e., middle 33%).

Reply:
Done